src/backend/joanie/core/models/products.py

```diff
@@ -977,6 +990,9 @@ def submit_for_signature(self, user: User):
             )
             raise PermissionDenied(message)
 
+        if not self.is_free:
+            self.generate_schedule()
```

Comment:
IMO it's still lacking a test asserting that the `payment_schedule` has changed once the method `submit_for_signature` has been called.